rfc/20/README.md

```diff
@@ -0,0 +1,84 @@
+
+## Vocabulary
+
+1. Mini-Provisioner API - a part of CORTX V2 Provisioner Framework. CORTX Provisioner defines Mini-Provisioner API interfaces to support phase-wise installation for each of the components which is specified using standard template: `/opt/seagate/cortx/<component>/conf/setup.yaml`
+2. ConfStore - a [lens-like thing](https://medium.com/@russmatney/haskell-lens-operator-onboarding-a235481e8fac) (yes, almost in the sense of Haskell) to work with various file formats and potentially databases as if it is a Key-Value storage. The idea is that different components may require similar knowledge about the cluster configuration or produce some knowledge that can be used later by another components during their self-deployment. So it is natural that at the level of Provisioner Framework we have a shared "database" to store and exchange the facts. But instead of the "database" we're given with an optics to it.
+
+## General setup.yaml structure
+
+```yaml
+    <component>:   
+      post_install:  				# Example: 
+        cmd: <script path>			# Script provided by Component
+        args: --config $URL			# ConfStore Config URL provided by Provisioner 
+      config:  			 
+        cmd: csm_setup				# Example: 
+        args: --cmd config			# Command and Args provided by Component
+      init:   
+        cmd: <script path> 
+        args: --config $URL 
+      test:   
+        cmd: <script path> 
+        args: --config $URL [--plan]		# Perform plan specific tests
+      upgrade:   
+        cmd: <script path> 
+        args: --config $URL 
+      reset:  					# Remove logs and meta-data (accounts) 
+        cmd: <script path> 			# Delete test related data 
+        args: --config $URL			 
+      cleanup:              			# Remove specific configuration (schema) 
+        cmd: <script path>                      # Delete config related data 
+        args: --config $URL 
+      backup: 					# URL provided by Component 
+        cmd: <script path> 
+        target: <URL>     			# Custom end-point to backup config (Consul) 
+      restore: 
+        cmd: <script path> 
+        target: <URL>           		# Custom end-point for restoring config (Consul) 
+      refresh_config:   
+        cmd: <script path> 			# To be handled by cleanup 
+        args: --config $URL 
+    support_bundle:   
+      - /opt/seagate/cortx/provisioner/cli/provisioner-bundler   
+```
+
+### Stage conditions
+
+| Stage        | Purpose           | Expected initial state  | Side effects performed as a result |
+| ------------- |-------------| ----- | --- |
+| post_install  | Verify that all the required 3rd party components are installed and have correct versions (Optional?) | Component RPM has just been installed to the OS. No services are started. | None (stage either fails or succeeds) |
```

Comment:
I think we need not re-iterate what is present in the `mini-provisioner` document but specifically mention what Hare will be doing in each section, e.g. config - generate CDF using information from conf-store.

Reply:
Yes, that will be covered in later sections of this document.